src/puppetlabs/trapperkeeper/services/webserver/jetty10_websockets.clj

```diff
@@ -77,39 +77,60 @@
   (log/debug (i18n/trs "No handler defined for websocket event ''{0}'' with args: ''{1}''"
                        event args)))
 
+(def client-count (atom 0))
+(defn extract-CN-from-certs
+  [x509certs]
+  (.getSubjectX500Principal (first x509certs)))
+
 (schema/defn ^:always-validate proxy-ws-adapter :- WebSocketAdapter
   [handlers :- WebsocketHandlers
    x509certs :- [X509Certificate]
    requestPath :- String
    closureLatch :- CountDownLatch]
-  (let [{:keys [on-connect on-error on-text on-close on-bytes]
+  (let [client-id (swap! client-count inc)
+        {:keys [on-connect on-error on-text on-close on-bytes]
          :or {on-connect (partial no-handler :on-connect)
               on-error   (partial no-handler :on-error)
               on-text    (partial no-handler :on-text)
               on-close   (partial no-handler :on-close)
               on-bytes   (partial no-handler :on-bytes)}} handlers]
     (proxy [WebSocketAdapter CertGetter ClosureLatchSyncer] []
       (onWebSocketConnect [^Session session]
+        (log/tracef "%d on-connect certname:%s uri:%s" client-id (extract-CN-from-certs x509certs) requestPath)
         (let [^WebSocketAdapter this this]
           (proxy-super onWebSocketConnect session))
-        (on-connect this))
+        (let [on-connect-result (on-connect this)]
+          (log/tracef "%d exiting on-connect certname:%s uri:%s" client-id (extract-CN-from-certs x509certs) requestPath)
```

Comment:
I think the `exit` messages can just include the client id, and the `exiting <behavior>` and not the extra arguments (they don't change between the start of the function and the end.

Reply:
The log lines will naturally have the thread identifier with them, so we should be able to pair them up.

Reply:
Seems reasonable, updated